### PR TITLE
refactor(exec-wasmtime): minor improvements

### PR DESCRIPTION
**Pull request by rvolosatovs**
A few commits extracted from https://github.com/enarx/enarx/pull/2174 to simplify review

### CODE REVIEW

File: crates/exec-wasmtime/src/loader/compiled/mod.rs [hunk 1]
```diff
@@ -20,15 +20,15 @@ impl Loader<Compiled> {
         let ctx = ctx.data_mut();
 
         // Set up environment variables.
-        for (k, v) in self.0.config.env.iter() {
```
Comment (bstrie) [inline crates/exec-wasmtime/src/loader/compiled/mod.rs:23]: Is this change just intended to get rid of the explicit `.iter()`, or is the intent to change this from by-ref iteration to by-val? If it's the former, you can do `for (k, v) in &self.0.config.env {` and avoid taking references inside the loop.

File: crates/exec-wasmtime/src/loader/compiled/mod.rs [hunk 2]
```diff
@@ -20,15 +20,15 @@ impl Loader<Compiled> {
...
-            ctx.push_env(k, v)?;
+        for (k, v) in self.0.config.env {
+            ctx.push_env(&k, &v)?;
         }
 
         // Set up the arguments.
         ctx.push_arg("main.wasm")
             .context("failed to push argv[0]")?;
-        for arg in self.0.config.args.iter() {
```
Comment (bstrie) [inline crates/exec-wasmtime/src/loader/compiled/mod.rs:30]: Same.

File: crates/exec-wasmtime/src/loader/compiled/mod.rs (hunk 1 above)
Comment (rvolosatovs) [reply crates/exec-wasmtime/src/loader/compiled/mod.rs:23]: The intention was to move it.

File: crates/exec-wasmtime/src/loader/compiled/mod.rs (hunk 2 above)
Comment (rvolosatovs) [reply crates/exec-wasmtime/src/loader/compiled/mod.rs:30]: Same intention, `self.0.config.args` is already processed, repeated usage of it should result in compile-time error.

**Review by haraldh — APPROVED** [review]

**Pull request merged**